implementation/test/TBTCSystemTest.js

```diff
@@ -67,6 +82,16 @@ describe("TBTCSystem", async function() {
         "TBTCSystem did not correctly forward value to keep factory",
       ).to.eq.BN(openKeepFee)
     })
+
+    it("reverts if caller does not match a valid TDT", async () => {
+      await expectRevert(
+        tbtcSystem.requestNewKeep(5, 10, 0, {
+          value: openKeepFee,
+          from: accounts[1],
```

Comment:
We should use at least two cases to validate the revert:
1. When a caller is a TDT owner.
2. When a caller is `accounts[0]`, that is neither the TDT owner nor the deposit (TDT ID).

Reply:
I'm not really clear on this, can you explain a little bit further?

> When a caller is a TDT owner

this should not revert.

Reply:
> We should use at least two cases to validate the revert:

Why? the test does not care about who holds the TDT, only if it exists.

Reply:
As I understand only Deposit contract can call `requestNewKeep` which is determined by TDT token ID. But TDT owner is a different account and we should fail to call `requestNewKeep` by the TDT owner. Am I incorrect?

Reply:
you're correct, but the test will not be checking any logic. We don't crae who the owner is and there is no owner related logic. We just care that the sender is addres(tokenId)

Reply:
Token owner is not used in the logic right now but it's very much related to the token and verification we perform. Having such a case can help us catching a regression bug in the future when someone updates e.g. token implementation in a way that impacts this verification.

Reply:
Added test to check if it will revert given a caller is the owner of a valid TDT.